### ISSUE TITLE
Add an option to generate JSDoc-style comments

### ISSUE THREAD

**Issue by yudai-nkt** [d-kimuson/type-predicates-generator#1]
Thank you for making this tool and I'm trying to use it in my projects! My projects use [typedoc](https://typedoc.org) to generate an API documentation ([example](https://yudai-nkt.github.io/switchbot-client/)), but the functions generated by type-predicates-generator lack their documentation because there is no corresponding comments. I'd appreciated it if such comments can be automatically made as well.

I've made a prototype of this feature at my [`jsdoc`](https://github.com/yudai-nkt/type-predicates-generator/tree/jsdoc) branch and am now struggling with format issues like indentations or blank lines. Before working further and sending a PR, I'd like to ask your opinion on this idea.

My initial motivation is a use case with typedoc, but other users will also benefit from this feature with the aid of LSP as shown below.

![type-predicates-generator-1637991917889](https://user-images.githubusercontent.com/13291527/143670172-8bbfd6fe-9598-4166-8f0f-c13eb0a6386b.gif)

**Comment by d-kimuson** [comment]
@yudai-nkt  Thanks for the feature suggestion and the prototype! I'm glad to hear that you're considering adopting it!

I think it's a good idea to add documentation with JSDoc comments, since the auto-generated code will remain under the project. I think the current policy of adding the option to add jsdoc is a good one, as shown in the [jsdoc branch](https://github.com/yudai-nkt/type-predicates-generator/tree/jsdoc).

As for indentation and formatting, currently it's difficult to adjust indentation due to using [dedent](https://github.com/dmnd/dedent), and I've touched with the jsdoc branch a bit and it seems to be difficult due to dedent as well. So I added a [commit](https://github.com/d-kimuson/type-predicates-generator/commit/b03437dd1e76103de894d9fff9de8ace63173f24) to change it so that it doesn't use dedent.I think this change will make it easier to make minor blank line and indent adjustments.

Thank you so much!
